app/index.js

```diff
@@ -210,9 +210,13 @@ var BackboneGenerator = yeoman.generators.Base.extend({
       this.indexFile = this.readFileAsString(path.join(this.sourceRoot(), 'index.html'));
       this.indexFile = this.engine(this.indexFile, this);
 
-      this.indexFile = this.appendScripts(this.indexFile, 'scripts/main.js', [
-        'bower_components/requirejs/require.js'
-      ], {'data-main': 'scripts/main'});
+      var attrs = this.attributes({
+        'data-main': 'scripts/main',
+        src: 'bower_components/requirejs/require.js'
+      });
+      var scriptTag = '    <script ' + attrs + '></script>\n';
```

Comment:
What did you have in mind @eddiemonge? I don't think there's a better way if we don't want to over-complicate it.